src/lib/kw_config_loader.sh

```diff
@@ -358,8 +358,11 @@ function parse_configuration()
   if [ ! -f "$config_path" ]; then
     return 22 # 22 means Invalid argument - EINVAL
   fi
+
+  # @see https://stackoverflow.com/questions/12916352/shell-script-read-missing-last-line
+  # for information about the need for the second check in the while loop.
   # shellcheck disable=SC2162
-  while read line; do
+  while read line || [ -n "$line" ]; do
```

Comment:
Use `[[ -n "$line" ]]` instead, as `[...]` is the "legacy" conditional construct.